main.cc

```diff
@@ -404,13 +399,13 @@ run(const char* prog_name)
       break;
     case 'x':
       optarg = FETCH_OPTARG;
-      filter = find_filter_vec(optarg);
+      filter = FilterVecs::Instance().find_filter_vec(optarg);
```

Comment:
Could filter just have a scope within this case and the 'free_filter_vec' be the destructor?

Reply:
Perhaps, but I think that would work better if find_filter_vec was the filter constructor.  At this point our formats and filters rely on state being preserved between uses (e.g. stack filter) so we aren't ready for dynamic filters/formats.